packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,335 @@
+  );
+  background-size: 600px;
+  animation: ${backgroundShimmer} 4s infinite linear;
+  transition: width 0.3s ease-in-out;
+`;
+
+interface ProgressBarProps {
+  progress: number;
+}
+
+function ProgressBar({ progress }: ProgressBarProps) {
+  return (
+    <div className={progressBarBackgroundStyle}>
+      <div
+        className={cx(
+          progressBarStyle,
+          css`
+            width: ${progress * 100}%;
+          `,
+        )}
+      />
+    </div>
+  );
+}
+
+const RTGStates = {
+  Entering: 'entering',
+  Entered: 'entered',
+  Exiting: 'exiting',
+  Exited: 'exited',
+} as const;
+
+type RTGStates = typeof RTGStates[keyof typeof RTGStates];
+
+const ToastVariants = {
+  success: 'success',
+  progress: 'progress',
+} as const;
+
+type ToastVariants = typeof ToastVariants[keyof typeof ToastVariants];
+
+const transitionDuration = 150;
+
+const baseToastStyle = css`
+  position: fixed;
+  bottom: ${spacing[6]};
+  left: ${spacing[4]};
+  width: ${toastWidth}px;
+  max-width: calc(100vw - (${spacing[4]} * 2));
+  border-radius: 4px;
+  box-shadow: 0 18px 18px -15px ${transparentize(0.7, uiColors.black)};
+  overflow: hidden;
+  transform: translate3d(0, ${spacing[3]}, 0) scale(0.95);
+  transform-origin: bottom center;
+  opacity: 0;
+  transition: all ${transitionDuration}ms ease-in-out;
+`;
+
+const toastVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    background-color: ${uiColors.green.light3};
+  `,
+
+  [ToastVariants.progress]: css`
+    background-color: ${uiColors.white};
+    padding-bottom: 6px;
+  `,
+};
+
+const toastVisibleStateStyle = css`
+  transform: translate3d(0, 0, 0) scale(1);
+  opacity: 1;
+`;
+
+const toastTransitionStateStyles: Partial<Record<RTGStates, string>> = {
+  [RTGStates.Entered]: toastVisibleStateStyle,
+};
+
+const baseIconStyle = css`
+  flex-shrink: 0;
+  margin-right: ${spacing[3]};
+`;
+
+const iconVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    color: ${uiColors.green.base};
+  `,
+
+  [ToastVariants.progress]: css`
+    color: ${uiColors.gray.dark2};
+  `,
+};
+
+const contentWrapperStyles = css`
+  display: flex;
+  align-items: center;
+  padding: ${spacing[3]};
+`;
+
+const contentWrapperVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    border-radius: 4px;
+    border: 1px solid ${uiColors.green.light2};
+  `,
+
+  [ToastVariants.progress]: css`
+    border-radius: 4px 4px 0 0;
+    border: 1px solid ${uiColors.blue.light2};
+    border-bottom: 0;
+  `,
+};
+
+const bodyVariantStyles: Record<ToastVariants, string> = {
+  [ToastVariants.success]: css`
+    color: ${uiColors.green.dark2};
+  `,
+
+  [ToastVariants.progress]: css`
+    color: ${uiColors.gray.dark2};
+  `,
+};
+
+const dismissButtonStyle = css`
+  position: absolute;
+  top: ${spacing[2]};
+  right: ${spacing[2]};
+  transition: color 0.15s ease-in-out;
+`;
+
+const dismissButtonVariantStyle = {
```

Comment:
nit: Recently added a variant to Callout and it was really easy to do because all of the possible states were in an object for me. It ensured that I didn't miss anything when trying to add a variant and also was super readable. Maybe this is a pattern we think about applying around when working with variants (specifically when those variant sets will probably grow)

``` [Variant.Note]: {
    header: {
      background: uiColors.blue.light3,
      text: uiColors.blue.dark2,
    },
    text: uiColors.blue.dark3,
    bar: uiColors.blue.base,
    icon: uiColors.blue.base,
  },```

Reply:
It's possible this is a better approach 🤔

Reply:
Yep, this definitely feels a lot better! Just updated this.

Reply:
👍